ui/js/search.js

```diff
@@ -0,0 +1,171 @@
+            }
+        }
+        $('.dialog').addClass('open');
+        dropdownOpen = true;
+        resetEventListeners(); 
+    }
+
+    function trackArrowKeysAndEnterBtn() {
+        $('.autocomplete').keydown(function (e) {
+            e.stopPropagation();
+
+            switch (e.which) {
+                case 13: // enter button 
+                    chooseOptionWithElement($('.dialog > div').eq(currentlySelected));
+                    break;
+                case 38: // up
+                    selectOptionWithIndex(-1);
+                    break;
+                case 40: // down
+                    selectOptionWithIndex(1);
+                    break;
+                default: return;
+            }
+        });
+    }
+
+    function selectOptionWithIndex(arrowDirection) {
+        if ((currentlySelected < 1 && arrowDirection < 1) ||
+            (currentlySelected === numOptions - 1 && arrowDirection > -1) ||
+            !dropdownOpen || disableArrowSelection) {
+            return;
+        }
+        var options = $('.dialog > div');
+        var currentSelection = options.eq(currentlySelected);
+        currentSelection.removeClass('selected');
+        currentSelection.removeAttr('id');
+        currentlySelected += arrowDirection;
+        startSelectOptionVisuals(options, arrowDirection);
+    }
+
+    function startSelectOptionVisuals(searchResults, arrowDirection) {
+        var newSelection = searchResults.eq(currentlySelected);
+        newSelection.addClass('selected');
+        newSelection.attr('id', 'selectedItem');
+
+        var scrollDivisor = 1.20;
+        if (arrowDirection == -1) {
+            scrollDivisor = 2;
+        }
+        $('.dialog').animate({
+            scrollTop: $("#selectedItem").offset().top / scrollDivisor
+        }, 100);
+        $('.autocomplete input').val(newSelection[0].innerHTML);
+        $('.autocomplete input').css('color', '#757575');
+    }
+
+    function toggleSearchResultsDialog(forceClose) {
+        $('.dialog > div').eq(currentlySelected).removeAttr("selectedItem");
+        if (forceClose) {
+            $('.dialog').removeClass('open');
+            dropdownOpen = false;
+        } else {
+            $('.dialog').toggleClass('open');
+            dropdownOpen = !dropdownOpen;
+        }
+    }
+
+    function chooseOptionWithElement(element) {
+        var courseId = element.attr("data-course-id");
+        if (chosenCourses.indexOf(courseId) < 0) {
+            chosenCourses.push(courseId);
+            console.log(chosenCourses);
+            $(".collection").append('<li class="collection-item" data-course-id=' + courseId + '>\
+            <div>'+ element[0].innerHTML + '\
+            <i class="material-icons deleteIcon">delete</i></div></li>');
+        }
+        $('.autocomplete input').val("").focus();
+        deleteCourseListener();
+
+        toggleSearchResultsDialog(true);
+    }
+
+    function deleteCourseListener() {
```

Comment:
Do you think `setDeleteCourseListener` would be a more descriptive function name?